src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -960,13 +973,49 @@ private void parseCommand() throws SnowflakeSQLException {
     // SNOW-15153: verify that the value after file:// is not changed by GS
     verifyLocalFilePath(localFilePathFromGS);
 
-    // more parameters common to upload/download
-    String stageLocation = jsonNode.path("data").path("stageInfo").path("location").asText();
-
     parallel = jsonNode.path("data").path("parallel").asInt();
 
     overwrite = jsonNode.path("data").path("overwrite").asBoolean(false);
 
+    stageInfo = getStageInfo(jsonNode);
+
+    if (logger.isDebugEnabled()) {
+      logger.debug("Command type: {}", commandType);
+
+      if (commandType == CommandType.UPLOAD) {
+        logger.debug("autoCompress: {}", autoCompress);
+
+        logger.debug("source compression: {}", sourceCompression);
+      } else {
+        logger.debug("local download location: {}", localLocation);
+      }
+
+      logger.debug("Source files:");
+      for (String srcFile : sourceFiles) {
+        logger.debug("file: {}", srcFile);
+      }
+
+      logger.debug("stageLocation: {}", stageInfo.getLocation());
```

Comment:
I know this is existing code, but it might be better to have one logger line that has all of this information in there as opposed to several log lines. I don't know if there's anything that would break doing that, so I'll defer to Meg on that one.

Reply:
I agree- it would be great if all these log lines could be combined into 1 with the values separated by commas.